src/morganJson.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Adopted from `morgan-json` from `indexzero` under MIT.
+ * https://github.com/indexzero/morgan-json/
+ * Originally adopted from `morgan.compile` from `morgan` under MIT.
+ * https://github.com/expressjs/morgan
+ */
+
+function isInvalidFormat(format) {
+    return !format
+      || format === ''
+      || (typeof format !== 'object'
+          && typeof format !== 'string');
+}
+
+function makeSafe(input) {
```

Comment:
I'd add some comments either here or in `generateBody` below detailing what/why/how this is making something safe -- IMO it's not immediately obvious.